src/Seq.Extensions.Logging/Seq/Extensions/Logging/JsonSafeString.cs

```diff
@@ -0,0 +1,30 @@
+﻿#nullable enable
+
+namespace Seq.Extensions.Logging
+{
+    /// <summary>
+    /// A wrapper type that marks a string as containing valid JSON that's safe to include as-is in log events.
+    /// </summary>
+    public readonly struct JsonSafeString
+    {
+        string? Json { get; }
+
+        /// <summary>
+        /// Construct a <see cref="JsonSafeString"/> with well-formed JSON. The JSON is not validated: it must be
+        /// externally validated or proven to be valid JSON before calling this constructor.
+        /// </summary>
+        /// <param name="json">A well-formed JSON string that can be included directly in a log event.</param>
+        public JsonSafeString(string json)
```

Comment:
IMO `JsonSafeString` should take a `string?` here. Since you're already checking for null in `.ToString()` it should be safe to pass a null string here.

Reply:
Thanks for the note, @danbopes! In this case the intention's to communicate that the constructor is handed well-formed JSON, we're only really handling the `null` case in `ToString()` because it's impossible to prevent it in C#.